### ISSUE TITLE
New linter: detect ClojureScript negative zero comparisons

### ISSUE THREAD

**Issue by marco-carvalho** [clj-kondo/clj-kondo#2965]
- [ ] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

In JavaScript, ordinary equality cannot distinguish negative zero from positive zero. ClojureScript code comparing a value with a negative-zero literal may therefore express an impossible distinction.

```clojure
(= x -0.0)
```

**Describe the solution you'd like**

Add a `:negative-zero-comparison` linter, enabled at warning level by default, for ClojureScript equality comparisons containing a syntactic negative-zero literal. Recommend `js/Object.is`. Do not report ordinary zero, nonzero negative values, or Clojure code.

**Describe alternatives you've considered**

A general numeric comparison warning would be too broad. Runtime checks with `js/Object.is` work, but the purpose is to identify equality calls that cannot provide the intended result.

**Additional context**

I am interested in submitting a focused PR with decimal and exponent forms if this approach is accepted.

**Comment by NoahTheDuke** [comment]
Is this worthy of a dedicated linter? Does it happen frequently?

**Comment by marco-carvalho** [comment]
I found no occurrences in the pinned regression corpora, so I do not have evidence that it is frequent. It is a real JavaScript correctness trap, but a dedicated default-on linter is probably disproportionate. Making it opt-in, folding it into a broader numeric-correctness linter, or closing it would all be reasonable.

**Comment by NoahTheDuke** [comment]
this rule does exist for eslint, which is a vote in favor: https://eslint.org/docs/latest/rules/no-compare-neg-zero
